protocol/peer/peer.go

```diff
@@ -0,0 +1,108 @@
+package peer
+
+import (
+	"github.com/kaspanet/kaspad/netadapter"
+	"github.com/kaspanet/kaspad/util/daghash"
+	"github.com/kaspanet/kaspad/util/subnetworkid"
+	"github.com/kaspanet/kaspad/wire"
+	"github.com/pkg/errors"
+	"sync"
+	"sync/atomic"
+)
+
+// Peer holds data about a peer.
+type Peer struct {
+	ready uint32
+
+	selectedTipHashMtx sync.RWMutex
+	selectedTipHash    *daghash.Hash
+
+	id                 uint32
+	userAgent          string
+	services           wire.ServiceFlag
+	advertisedProtoVer uint32 // protocol version advertised by remote
+	protocolVersion    uint32 // negotiated protocol version
+	disableRelayTx     bool
+	subnetworkID       *subnetworkid.SubnetworkID
+}
+
+// SelectedTipHash returns the selected tip of the peer.
+func (p *Peer) SelectedTipHash() (*daghash.Hash, error) {
+	if atomic.LoadUint32(&p.ready) == 0 {
+		return nil, errors.New("peer is not ready yet")
+	}
+	p.selectedTipHashMtx.RLock()
+	defer p.selectedTipHashMtx.RUnlock()
+	return p.selectedTipHash, nil
+}
+
+// SubnetworkID returns the subnetwork the peer is associated with.
+// It is nil in full nodes.
+func (p *Peer) SubnetworkID() (*subnetworkid.SubnetworkID, error) {
+	if atomic.LoadUint32(&p.ready) == 0 {
+		return nil, errors.New("peer is not ready yet")
+	}
+	return p.subnetworkID, nil
+}
+
+// SetSelectedTipHash sets the selected tip of the peer.
```

Comment:
It could be nice if we didn't have SubnetworkID() between this and SelectedTipHash